src/choose_gossip_peer_strategy.rs

```diff
@@ -0,0 +1,306 @@
+use crdt::ReplicatedData;
+use rand::distributions::{IndependentSample, Weighted, WeightedChoice};
+use rand::thread_rng;
+use result::{Error, Result};
+use signature::PublicKey;
+use std;
+use std::collections::HashMap;
+
+pub const DEFAULT_WEIGHT: u32 = 1;
+
+pub trait ChooseGossipPeerStrategy {
+    fn choose_peer(&self, options: Vec<&ReplicatedData>) -> Result<ReplicatedData>;
```

Comment:
Are you able to return a reference here to avoid the `clone()`?

Reply:
Also, @aeyakovenko, what do you think about the use of the term "peer" here? Should we consider replacing `ReplicatedData` with `Peer`?

Reply:
@garious, yeah I thought the reference would have been fine, but in the original gossip_request() code, there was a clone, so I aligned with that. https://github.com/solana-labs/solana/blob/master/src/crdt.rs#L520. Looking at it more in depth now, it doesn't seem necessary.